src/pykeen/evaluation/ogb_evaluator.py

```diff
@@ -0,0 +1,142 @@
+"""OGB tools."""
+
+import logging
+from typing import Dict, Iterable, List, Optional, Tuple
+
+import torch
+
+from .evaluator import MetricResults
+from .rank_based_evaluator import RankBasedMetricResults, SampledRankBasedEvaluator
+from .ranking_metric_lookup import MetricKey
+from ..metrics import RankBasedMetric
+from ..metrics.ranking import HitsAtK, InverseHarmonicMeanRank
+from ..models import Model
+from ..typing import RANK_REALISTIC, SIDE_BOTH, ExtendedTarget, MappedTriples, RankType, Target
+
+__all__ = [
+    "evaluate_ogb",
+]
+
+logger = logging.getLogger(__name__)
+
+
+def evaluate_ogb(
```

Comment:
one comment before we merge - could we consider subclassing `SampledRankBasedEvaluator` as `OGBEvaluator` and just override its evaluation code with this?

Reply:
Started doing this in abc6137